### ISSUE TITLE
No certificates generated with acme.json from traefik v2.0

### ISSUE THREAD

**Issue by rezor92** [ldez/traefik-certs-dumper#46]
Hi, 

first of all thanks for your work and that you are sharing it. I have tried it but I cannot get it to work on my server.  

I'm using _traefik v2.0_ and the docker image _ldez/traefik-certs-dumper:latest_ with the following command:

```
  certdumper:
    image: ldez/traefik-certs-dumper:latest
    command: "file --watch --domain-subdir=true --version v2"
    volumes:
      - "/XX/XX/XX/acme.json:/acme.json:ro"
      - "/XX/XX/certs:/dump"
```

I'm getting the following log after the acme.json changed:

> 2019/09/25 07:15:54 file.go:96: error: EOF,
> dump,
> └──private,

If I check the output folders they are empty. It would be kind if someone could give me a hint what I'm doing wrong.

Best regards

_Originally posted by @rezor92 in https://github.com/ldez/traefik-certs-dumper/issues/34#issuecomment-534892356_

**Comment by straend** [comment]
Other similiar tools also have the same problem, there seems to be a new format inside the `acme.json` storage.
https://github.com/DanielHuisman/traefik-certificate-extractor/issues/18

**Comment by timoschwarzer** [comment]
Confirming that no certificated are being extracted with an `acme.json` from Traefik 2.0.1

**Comment by timoschwarzer** [comment]
@ldez seems like it was this commit (https://github.com/containous/traefik/commit/c7d336f958f04ed63a76e308317eb5b196e01ad0#diff-153848c6b668f0e912ccbef254a4e23c) that broke the dumper, which still uses Title-Cased keys :P

**Comment by ldez** [comment]
I don't think so because I use the [v1 structures](https://github.com/ldez/traefik-certs-dumper/blob/master/dumper/v1/storeddata.go) and the v2 structures come from traefik it-self.

**Comment by ldez** [comment]
@timoschwarzer thanks.

Fixed in v2.5.5

**Comment by rezor92** [comment]
> @timoschwarzer thanks.
> 
> Fixed in v2.5.5
@ldez Thank you. @timoschwarzer  Is it working for you? For me using the newest version still doesn't resolve my issue. Only empty private folder :(

**Comment by timoschwarzer** [comment]
@rezor92 @ldez I just tried it, and unfortunately you are right. I'm using the latest Docker image 8e578f66d86d.

**Comment by timoschwarzer** [comment]
I'm not really familiar with Go but I think the problem is that the v2 dumper doesn't take multiple certificate resolvers into account.

My acme.json looks like this:

```json
{
  "<certResolverName>": {
    "Account": { ... },
    "Certificates": [
      {"domain": ..., "certificate": ..., "key": ..., "Store": ...},
      ...
    ]
  }
}
```

It looks like traefik-certs-dumper searches for the `Certificates` key in the root JSON-Object.

**Comment by ldez** [comment]
Fixed in v2.5.6.

I will add some non regression tests.

**Comment by timoschwarzer** [comment]
@ldez Unfortunately, `traefik-certs-dumper file --version v2` still does not work for me. I'll try to investigate why.

**Comment by timoschwarzer** [comment]
There seems to be revision pinned that no longer exists (`go: github.com/labbsr0x/goh@v0.0.0-20190417202808-8b16b4848295: unknown revision 8b16b4848295`) so unfortunately I cannot build and test traefik-certs-dumper. (Unless there's a workaround for that...)

**Comment by ldez** [comment]
I tested with real acme.json file without any issues.

Could you give me more information?

---

I recommend to use go1.13.

**Comment by timoschwarzer** [comment]
@ldez 

> I recommend to use go1.13.

I installed Go 1.13. I still cannot install all dependencies:

```
go: github.com/go-acme/lego/v3@v3.0.2 requires
	github.com/labbsr0x/goh@v0.0.0-20190417202808-8b16b4848295: invalid version: unknown revision 8b16b4848295
```

> Could you give me more information?

Of course:

### Traefik version
```
Version:      2.0.1
Codename:     montdor
Go version:   go1.13.1
Built:        2019-09-26T16:18:03Z
OS/Arch:      linux/amd64
```

### My acme.json (redacted)
```json
{
  "default": {
    "Account": {
      "Email": "me@email.com",
      "Registration": {
        "body": {
          "status": "valid",
          "contact": [
            "mailto:me@email.com"
          ]
        },
        "uri": "https://acme-v02.api.letsencrypt.org/acme/acct/12345678"
      },
      "PrivateKey": "FOOOBAR",
      "KeyType": "4096"
    },
    "Certificates": [
      {
        "domain": {
          "main": "my.domain.com"
        },
        "certificate": "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCkZPT09PT09PTwotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCgotLS0tLUJFR0lOIENFUlRJRklDQVRFLS0tLS0KRk9PT08KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=",
        "key": "LS0tLS1CRUdJTiBSU0EgUFJJVkFURSBLRVktLS0tLQpCQUFBQUFBQVIKLS0tLS1FTkQgUlNBIFBSSVZBVEUgS0VZLS0tLS0K",
        "Store": "default"
      },
      {
        "domain": {
          "main": "my.domain2.com"
        },
        "certificate": "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCkZPT09PT09PTwotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCgotLS0tLUJFR0lOIENFUlRJRklDQVRFLS0tLS0KRk9PT08KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=",
        "key": "LS0tLS1CRUdJTiBSU0EgUFJJVkFURSBLRVktLS0tLQpCQUFBQUFBQVIKLS0tLS1FTkQgUlNBIFBSSVZBVEUgS0VZLS0tLS0K",
        "Store": "default"
      }
    ]
  }
}
```

### Traefik certificate resolver config
```yaml
certificatesResolvers:
  default:
    acme:
      email: me@redacted.com
      storage: /acme.json
      keyType: RSA4096
      tlsChallenge: {}
```

### Command

I tried version 2.5.6 from both GitHub releases and the Docker image.

```
$> ./traefik-certs-dumper file --version v2
dump
├──certs
└──private
```

**Comment by ldez** [comment]
I found the issue.

**Comment by ldez** [comment]
Sorry for the multiple releases, I'm on vacation so I did it a little fast.

Fixed in v2.5.7.

**Comment by timoschwarzer** [comment]
No need to apologize! It works like a charm now, thank you for your time! :)

Have a nice vacation! :relaxed:

**Comment by rezor92** [comment]
> Sorry for the multiple releases, I'm on vacation so I did it a little fast.
> 
> Fixed in v2.5.7.

It's working now. Thank you so much for fixing it.
